homeassistant/components/powerwall/__init__.py

```diff
@@ -258,30 +258,30 @@ async def get_backup_reserve_percentage(power_wall: Powerwall) -> Optional[float
 
 async def _fetch_powerwall_data(power_wall: Powerwall) -> PowerwallData:
     """Process and update powerwall data."""
-    (
-        backup_reserve,
-        charge,
-        site_master,
-        meters,
-        grid_services_active,
-        grid_status,
-    ) = await asyncio.gather(
-        get_backup_reserve_percentage(power_wall),
-        power_wall.get_charge(),
-        power_wall.get_sitemaster(),
-        power_wall.get_meters(),
-        power_wall.is_grid_services_active(),
-        power_wall.get_grid_status(),
-    )
 
-    return PowerwallData(
-        charge=charge,
-        site_master=site_master,
-        meters=meters,
-        grid_services_active=grid_services_active,
-        grid_status=grid_status,
-        backup_reserve=backup_reserve,
-    )
+    try:
+        async with asyncio.TaskGroup() as tg:
```

Comment:
https://developers.home-assistant.io/docs/creating_component_code_review/#5-make-your-pull-request-as-small-as-possible

Please do the task group refactor in a separate PR